### ISSUE TITLE
Case when permission is canceled was not handled

### ISSUE THREAD

**Issue by arifrgilang** [anggrayudi/SimpleStorage#125]
Library version: 1.4.0
OS version: Android 13
Device model: Redmi Note 10 Pro

**Describe the bug**
Based on this documentation: https://developer.android.com/training/permissions/requesting#manage-request-code-yourself
There are certain cases where the permission interaction is canceled, resulting empty result.
<img width="679" alt="image" src="https://github.com/anggrayudi/SimpleStorage/assets/36944464/4b44654f-f3cd-45cd-a8b1-bde72fc593c9">
When cancelation occured and returns empty result, the `callback.onPermissionChecked()` (callback for granted permission) was called. This is unexpected (or at least isn't quite right) because no permisison is granted, yet the callback for granted permission is triggered.
![image](https://github.com/anggrayudi/SimpleStorage/assets/36944464/4b12e815-a431-4324-8bb2-d19f371ec576)

**To Reproduce**
I don't have the steps to reproduce it, but hope this screenshot when I debug it will help
![image](https://github.com/anggrayudi/SimpleStorage/assets/36944464/b82b0922-a65c-409c-9a50-b43f66b7b214)
In that screenshot, the result is empty and the `callback.onPermissionChecked()` block code is executed

**Comment by arifrgilang** [comment]
Since this cancelation behavior is quite rare, I suggest to add a callback too when `onPermissionIsCanceledOrInterrupted()` and have the default callback to call the `onShouldRedirectToSystemSettings()` if the method isn't provided by the developer that used the library

**Comment by anggrayudi** [comment]
I'm unable to get empty `result` on emulator, either by pressing back button nor tapping outside the permission request dialog. Please provide the steps to reproduce that empty map, so that I can proceed the bugfixing. I wonder if it's a device specific issue, like it is only happens on Xiaomi.

**Comment by arifrgilang** [comment]
@anggrayudi I am using camera permission inside Fragment.
Unfortunately, I still can't find the steps to reproduce it until now. My hypothesis is, the cancelation happened when you request permission while requesting permission.
```
val cameraPermission = FragmentPermissionRequest.Builder(this)
            .withPermissions(ManifestPermission.CAMERA)
            .withCallback((permissionResult, b) -> {
                if (permissionResult.getAreAllPermissionsGranted()) {
                    ScannerFragment.this.onPermissionGranted();
                } else {
                    onBack();
                }
            }).build();

cameraPermission.check();
```

**Comment by anggrayudi** [comment]
Please try this version, does it solve your problem?
```gradle
implementation 'com.anggrayudi:storage:1.5.5-SNAPSHOT'
```

**Comment by anggrayudi** [comment]
Released [v1.5.5](https://github.com/anggrayudi/SimpleStorage/releases/tag/1.5.5), and I've added `onPermissionRequestInterrupted()`
